src/common/ColourPalette.cpp

```diff
@@ -38,10 +50,342 @@ uInt32 ColourPalette::getRGB(int val) const {
     return m_palette[val];
 }
 
+const uInt32 *ColourPalette::getPalette()
+{
+    return m_palette;
+}
+
+void ColourPalette::setPalette(const string& type,
+                               const string& displayFormat)
+{
+    // See which format we should be using
+    int paletteNum = 0;
+    if(type == "standard")
+        paletteNum = 0;
+    else if(type == "z26")
+        paletteNum = 1;
+    else if(type == "user" && myUserPaletteDefined)
+        paletteNum = 2;
+
+    int paletteFormat = 0;
+    if (displayFormat.compare(0, 3, "PAL") == 0)
+        paletteFormat = 1;
+    else if (displayFormat.compare(0, 5, "SECAM") == 0)
+        paletteFormat = 2;
+
+    m_palette  = availablePalettes[paletteNum][paletteFormat];
+}
+
+void ColourPalette::calculateGrayscaleValues()
+{
+    for (int type = 0; type < 3; type++)
```

Comment:
This a minor point, but this code isn't threadsafe at the moment. Specifically: two threads could currently try to write to currentPalette. It may be worthwhile, at least, to test whether the grayscale values have been initialized, and not set them again. (Also, one could note that these grayscale values aren't going to change, so we could hardcode them).